client/internal/engine.go

```diff
@@ -766,6 +766,10 @@ func (e *Engine) addNewPeer(peerConfig *mgmProto.RemotePeerConfig) error {
 
 		go e.connWorker(conn, peerKey)
 	}
+	err := e.statusRecorder.UpdatePeerFQDN(peerKey, peerConfig.Fqdn)
```

Comment:
What will happen if fqdn is not there (e.g., old server

Also, what will happen if fqdn changes, will it be updated?

Reply:
> What will happen if fqdn is not there (e.g., old server

grpc method will return an empty string which is ok

> Also, what will happen if fqdn changes, will it be updated?

Good point. I pushed a fix. Basically, we will always update the peer's FQDN when they won't follow the modify process.